src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -687,7 +687,7 @@ class RNNOpRewriteLayoutPattern : public OpRewritePattern<ONNXOp> {
   LogicalResult matchAndRewrite(
       ONNXOp onnxOp, PatternRewriter &rewriter) const override {
     if (onnxOp.getLayout() == 0) {
-      return success();
+      return failure();
```

Comment:
This is a bug causing infinite loop in applying this rule, and the pass terminated without any error, which causes other rules have no chance to be applied. It must be `failure` when the pattern does not match.